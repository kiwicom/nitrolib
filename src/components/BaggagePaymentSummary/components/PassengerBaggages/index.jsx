// @flow
import * as React from "react";
import Stack from "@kiwicom/orbit-components/lib/Stack";
import Text from "@kiwicom/orbit-components/lib/Text";

import Price from "../../../Price";
import { getTextFromCategory } from "../../../../services/baggage/utils";
import type { ItemType } from "../../../../records/Baggage";
import Translate from "../../../Translate";

type Props = {
  firstName: string,
  lastName: string,
  baggage: {
    handBag: Array<ItemType>,
    holdBag: Array<ItemType>,
  },
  price: number,
};

const PassengerBaggages = ({ firstName, lastName, baggage, price }: Props) => (
  <Stack spaceAfter="medium" spacing="tight">
    <Stack flex justify="between">
      <Text>
        <Translate t="baggage_modal.summary.baggage_for" values={{ firstName, lastName }} />
      </Text>
      <Text>
        <Price value={price} />
      </Text>
    </Stack>
    <Stack spacing="tight">
      {baggage.handBag.map((bag, index) => (
        <Stack key={index} /* eslint-disable-line */>
          <Text type="secondary" element="span">
            {`${bag.amount}× `}
            {getTextFromCategory(bag.category)}
          </Text>
        </Stack>
      ))}
      {baggage.holdBag.map((bag, index) => (
        <Stack key={index} /* eslint-disable-line */>
          <Text type="secondary" element="span">
            {`${bag.amount}× `}
            {getTextFromCategory(bag.category)}
          </Text>
        </Stack>
      ))}
    </Stack>
  </Stack>
);

export default PassengerBaggages;
