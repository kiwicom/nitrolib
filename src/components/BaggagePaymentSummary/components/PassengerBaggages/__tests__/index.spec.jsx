// @flow strict
import * as React from "react";
import { shallow } from "enzyme";

import PassengerBaggages from "..";

const props = {
  paxId: 1,
  firstName: "Barrack",
  lastName: "Obama",
  baggage: {
    holdBag: [
      {
        category: "personalItem",
        restrictions: {
          weight: 3,
          height: 20,
          width: 20,
          length: 40,
          dimensionsSum: null,
        },
        amount: 1,
        conditions: {
          passengerGroups: ["adult"],
        },
      },
    ],

    handBag: [
      {
        category: "holdBag",
        restrictions: {
          weight: 10,
          height: 40,
          width: 30,
          length: 50,
          dimensionsSum: null,
        },
        amount: 1,
        conditions: {
          passengerGroups: ["adult"],
        },
      },
      {
        category: "holdBag",
        restrictions: {
          weight: 15,
          height: 50,
          width: 30,
          length: 50,
          dimensionsSum: null,
        },
        amount: 2,
        conditions: {
          passengerGroups: ["adult"],
        },
      },
    ],
  },
  price: 40,
};

describe("#BaggagePaymentSummary", () => {
  test("renders", () => {
    const wrapper = shallow(<PassengerBaggages {...props} />);
    expect(
      wrapper
        .find("Stack")
        .first()
        .prop("dataTest"),
    ).toBe("BaggagePaymentSummary-PassengerBaggages-1");

    expect(wrapper.find("Price").prop("value")).toBe(40);
  });
});
