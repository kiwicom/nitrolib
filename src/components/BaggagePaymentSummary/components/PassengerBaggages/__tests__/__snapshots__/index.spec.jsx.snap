// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#BaggagePaymentSummary renders 1`] = `
<Stack
  spaceAfter="medium"
  spacing="tight"
>
  <Stack
    flex={true}
    justify="between"
  >
    <Text>
      <Translate
        html={false}
        t="baggage_modal.summary.baggage_for"
        transform={[Function]}
        values={
          Object {
            "firstName": "Barrack",
            "lastName": "Obama",
          }
        }
      />
    </Text>
    <Text>
      <Price
        value={40}
      />
    </Text>
  </Stack>
  <Stack
    spacing="tight"
  >
    <Stack
      key="0"
    >
      <Text
        element="span"
        type="secondary"
      >
        1× 
        <Translate
          html={false}
          t="baggage_modal.select.checked_bag"
          transform={[Function]}
          values={Object {}}
        />
      </Text>
    </Stack>
    <Stack
      key="1"
    >
      <Text
        element="span"
        type="secondary"
      >
        2× 
        <Translate
          html={false}
          t="baggage_modal.select.checked_bag"
          transform={[Function]}
          values={Object {}}
        />
      </Text>
    </Stack>
    <Stack
      key="0"
    >
      <Text
        element="span"
        type="secondary"
      >
        1× 
        <Translate
          html={false}
          t="baggage_modal.select.personal_item"
          transform={[Function]}
          values={Object {}}
        />
      </Text>
    </Stack>
  </Stack>
</Stack>
`;
